docs/src/modules/components/AppDrawerNavItem.js

```diff
@@ -82,9 +82,15 @@ class AppDrawerNavItem extends React.Component {
       return (
         <ListItem className={classes.itemLeaf} disableGutters {...other}>
           <Button
-            component={props => (
-              <Link variant="button" activeClassName={classes.active} href={href} {...props} />
-            )}
+            component={React.forwardRef((props, ref) => (
```

Comment:
What do you think of moving the component outside of the render method?
I don't think that we should encourage creating new components in the render method.

Reply:
Probably better yes. I will need to look into this again though. I'm having a hard time wrapping my head around when I have to use `forwardRef` and when `innerRef` because of higher-order components. Especially our `Link` component which uses `withRouter` and `withStyles`. 

It's currently not working correctly. It might be a better idea to start with `forwardRef` in `withStyles`. It's also technically a breaking change (so targeted at `next`) but should make work on other components easier.